pkg/cli/admin/nodeimage/create.go

```diff
@@ -0,0 +1,578 @@
+
+	"github.com/spf13/cobra"
+	"k8s.io/klog/v2"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	kapierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	kutils "k8s.io/client-go/util/exec"
+	"k8s.io/kubectl/pkg/cmd/exec"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/library-go/pkg/operator/resource/retry"
+	imagemanifest "github.com/openshift/oc/pkg/cli/image/manifest"
+)
+
+const (
+	nodeJoinerConfigurationFile = "nodes-config.yaml"
+	nodeJoinerContainer         = "node-joiner"
+)
+
+func NewCreate(f kcmdutil.Factory, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewCreateOptions(streams)
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create an ISO image for booting the nodes to be added to the target cluster",
+		Long: templates.LongDesc(`
+			<TODO>
+		`),
+		Example: templates.Examples(`
+			<TODO>
+		`),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Validate())
+			kcmdutil.CheckErr(o.Run(cmd.Context()))
+		},
+	}
+
+	flags := cmd.Flags()
+	o.SecurityOptions.Bind(flags)
+
+	flags.StringVar(&o.AssetsDir, "dir", o.AssetsDir, "The path containing the configuration file, used also to store the generated artifacts.")
+	flags.StringVarP(&o.OutputName, "output-name", "o", "node.iso", "The name of the output image.")
+	return cmd
+}
+
+func NewCreateOptions(streams genericiooptions.IOStreams) *CreateOptions {
+	return &CreateOptions{
+		IOStreams: streams,
+	}
+}
+
+type CreateOptions struct {
+	genericiooptions.IOStreams
+	SecurityOptions imagemanifest.SecurityOptions
+
+	Config         *rest.Config
+	Client         kubernetes.Interface
+	ConfigClient   configclient.Interface
+	FSys           fs.FS
+	RemoteExecutor exec.RemoteExecutor
+	CopyStrategy   func(*rsync.RsyncOptions) rsync.CopyStrategy
+
+	AssetsDir  string
+	OutputName string
+
+	factory                  kcmdutil.Factory
+	nodeJoinerImage          string
+	nodeJoinerNamespace      *corev1.Namespace
+	nodeJoinerServiceAccount *corev1.ServiceAccount
+	nodeJoinerRole           *rbacv1.ClusterRole
+	nodeJoinerPod            *corev1.Pod
+	nodeJoinerExitCode       int
+	rsyncRshCmd              string
+}
+
+func (o *CreateOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	o.factory = f
+
+	var err error
+	if o.Config, err = f.ToRESTConfig(); err != nil {
+		return err
+	}
+	if o.Client, err = kubernetes.NewForConfig(o.Config); err != nil {
+		return err
+	}
+	if o.ConfigClient, err = configclient.NewForConfig(o.Config); err != nil {
+		return err
+	}
+
+	if o.AssetsDir == "" {
+		cwd, err := os.Getwd()
```

Comment:
I believe that there is corresponding wrapped function in kubernetes and no need to use `os.Getwd`

Reply:
:thinking: I wasn't aware, which one? I noticed that other commands like [login](https://github.com/openshift/oc/blob/a2e5b09d7d14b77c14f1b32550f8c5a0328e73b3/pkg/cli/login/loginoptions.go#L535) are using `os.Getwd`

Reply:
I'm just speculating, if it is used in oc in somewhere, feel free to use it :)